portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,48 @@
+package com.google.sps.servlets;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.util.concurrent.TimeUnit;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet
+{
+    @Override
+    public void doGet (HttpServletRequest request, HttpServletResponse response) throws IOException
+    {   
+        //Sets response type to html. We are writing directly to the page.
+        response.setContentType("text/html");
+        //Create UserService instance
+        UserService userService = UserServiceFactory.getUserService();
+        //if user is logged in we want to redirect back to index.html
+        if(userService.isUserLoggedIn())
+        {
+            String userEmail = userService.getCurrentUser().getEmail();
+            response.getWriter().println("<h1> You are already logged into " + userEmail+ "</h1>");
+            String logoutUrl = userService.createLogoutURL("/index.html");
+            response.getWriter().println("<p><a href=\"" + logoutUrl + "\">Logout</a>.</p>");
+        }
+        else //If user is not logged in prompt them to do so.
+        {
+            // To log someone in we must have a place to go to after login
+            String afterLoginRedirect = "/index.html";
+            //Now that we have somewhere to go to after login we can create loginURl
+            String loginUrl = userService.createLoginURL(afterLoginRedirect);
```

Comment:
Is this supposed to be in `Week2`? This looks like a library, which would be `Week3` :)

Reply:
Yea this is the first part of week3

Reply:
Yea this is the first part of week3

Reply:
Ah, since this PR was still open (I had some comments about your redirect URL that I don't think should be merged) the new commits to the branch show here. You can create a new branch from here and work off of that.

On your `week2` branch, run `git checkout -b week3`. This will create and checkout a new week3 branch off of the tip of `week2`. It also has the side effect of setting the `upstream` to `week2`, which we can fix later.

Then push your commit and create a pull request that is the diff between `week3` and `week2` branches. Usually you'd diff against `master` here.

Now you have 2 PRs.

Once `week2` is merged, we need to change `week3`'s ``upstream` to `master`. That's just `git branch -u origin/master`

If this is too much new content to handle right now, we can do this over hangouts later today or tomorrow afternoon.